src/clj/game/core/runs.clj

```diff
@@ -125,9 +125,11 @@
              {:prompt (str "Pay " trash-cost "[Credits] to trash " name "?")
               :yes-ability {:cost [:credit trash-cost]
                             :delayed-completion true
-                            :effect (effect (trash eid card nil)
-                                            (system-msg (str "pays " trash-cost
-                                                             " [Credits] to trash " name)))}}}
+                            :effect (req (trash state side eid card nil)
+                                         (let [src (name-zone :corp (:zone card))]
+                                           (system-msg state side (str "pays " trash-cost
+                                                                       " [Credits] to trash " name
+                                                                       (when-not (nil? src) (str " from " src))))))}}}
```

Comment:
I think `"in the root of" src` would be more correct for the centrals. But this way makes it compatible with remote server right?